client/src/Movies/MovieList.js

```diff
@@ -24,31 +26,32 @@ export default class MovieList extends Component {
     return (
       <div className="movie-list">
         {this.state.movies.map(movie => (
-          <MovieDetails key={movie.id} movie={movie} />
+          <div onClick={() => this.props.history.push(`/movies/${movie.id}`)} key={movie.id}><MovieDetails  movie={movie} /></div>
```

Comment:
You could just use a Link component here with this data on the `to` attribute.  The history object is useful for when you want to redirect and some other functionality (like an onSubmit) at the same time.